### PR TITLE
Update forcing generator

### PR DESCRIPTION
**Pull request by stefsmeets**
Hi all, this PR tunes the forcing generator functions to:

1. Return a `ForcingData` object, which at present is a very shallow wrapper around the esmvalcore `RecipeOutput`
2. Rename `forcings` -> `dataset` as an input parameter
3. Take a single dataset instead of a list

### CODE REVIEW

**Comment by Peter9192** [comment]
Hey @stefsmeets can you try to merge master? I think the CI was fixed (in #23) since you started working on this. Would be nice to see some green tests

File: ewatercycle/forcing_data.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class ForcingData(object):
```
Comment (Peter9192) [inline ewatercycle/forcing_data.py]: Maybe a dumb question, but why put this in a separate file? If I understand it correctly this leads to the following hierarchy (e.g. in the documentation)
```
ewatercycle 
- forcing
  - generate
- forcing_data
  - ForcingData
```

Wouldn't it make more sense to have something like:

```
ewatercycle
- forcing
  - ForcingData
  - generate
```
?

**Review by Peter9192 — CHANGES_REQUESTED** [review]
Looks good! Perhaps you could merge master to see if the tests pass, and maybe also merge #31 so we can preview the documentation build? Other than that I just have one question about how to structure the module.

File: ewatercycle/forcing_data.py (hunk 1 above)
Comment (stefsmeets) [reply ewatercycle/forcing_data.py]: `forcing.py` is already quite big, and I also expect `ForcingData` will not turn out to be a small class, so I put it in a different file. Besides that, it also gives some nice separation of concerns.

Later on we can move both files in to a sub directory to keep them together, i.e. `./ewatercycle/forcing/`

Comment (Peter9192) [reply ewatercycle/forcing_data.py]: Indeed, I like that last suggestion. Why not do that right away?

Comment (Peter9192) [reply ewatercycle/forcing_data.py]: Notice that Stefan also made a `models` module in #35

Comment (stefsmeets) [reply ewatercycle/forcing_data.py]: Restructured the code a bit in the latest commit 👍

**Review by Peter9192 — APPROVED** [review]

**Comment by Peter9192** [comment]
Thanks Stef!

**Pull request merged**